apps/examples/testcase/le_tc/kernel/tc_sched.c

```diff
@@ -524,6 +558,54 @@ static void tc_sched_sched_getstreams(void)
 	TC_SUCCESS_RESULT();
 }
 
+/**
+ * @fn                   :tc_sched_task_setcanceltype
+ * @brief                :This tc tests tc_sched_task_setcanceltype()
+ * @Scenario             :The task_setcanceltype() function atomically both sets the calling
+ *                        task's cancelability type to the indicated type and returns the
+ *                        previous cancelability type at the location referenced by oldtype
+ *                        If successful pthread_setcanceltype() function shall return zero;
+ *                        otherwise, an error number shall be returned to indicate the error.
+ * @API'scovered         :task_setcanceltype
+ * @Preconditions        :none
+ * @Postconditions       :none
+ * @return               :void
+ */
+#ifdef CONFIG_CANCELLATION_POINTS
+static void tc_sched_task_setcanceltype(void)
+{
+	pid_t pid;
+	int ret_chk;
+	int status = 1;
+
+	/* creating new process */
+	pid = task_create("sched1", SCHED_PRIORITY_DEFAULT, TASK_STACKSIZE, function_task_setcanceltype, (char * const *)NULL);
+	TC_ASSERT_GT("task_create", pid, 0);
+
+	ret_chk = waitpid(pid, &status, 0);
+	TC_ASSERT_EQ("waitpid", status, 1);	/* If any case fails, status value will be updated to -1 */
```

Comment:
Total_fail is increased by TC_ASSERT_XX inside function_task_setcanceltype. This makes a number of failure increasing twice.

Reply:
Let's change it like below.
```
(void)waitpid(pid, &status, 0);
if (status == OK) {
	TC_SUCCESS_RESULT();
}

Reply:
Oh, in success case, does it return 1? not 0?